### PR TITLE
refactor(ForumTopic): drop Author and read from author_id

### PR DESCRIPTION
**Pull request by wescopeland**
This PR drops the `Author` column from the `ForumTopic` table and switches all reads to instead use `author_id`.

Additionally, many places that were fetching a forum topic actually only needed to do an existence check. This has been optimized.

Several related helper functions have been migrated to Eloquent ORM.

### CODE REVIEW

File: app/Helpers/database/user-permission.php [hunk 1]
```diff
@@ -154,40 +154,38 @@ function getUserForumPostAuth(string $user): bool
...
+    $targetUser->ManuallyVerified = 1;
+    $targetUser->forum_verified_at = now();
+    $targetUser->save();
 
-    if (getAccountDetails($user, $userData)) {
-        addArticleComment('Server', ArticleType::UserModeration, $userData['ID'],
-            $sourceUser . ' authorized user\'s forum posts'
-        );
-    }
+    authorizeAllForumPostsForUser($targetUser->User);
```
Comment (Jamiras) [inline app/Helpers/database/user-permission.php:184]:
```
authorizeAllForumPostsForUser(): Argument #1 ($user) must be of type App\Models\User, string given, called in /var/www/html/app/Helpers/database/user-permission.php on line 184
```

File: resources/views/pages-legacy/forumposthistory.blade.php [hunk 2]
```diff
@@ -58,12 +59,18 @@
...
+    $fetchedUserCache = [];
+
     foreach ($recentPosts as $topicPostData) {
         $postMessage = $topicPostData['ShortMsg'];
         if ($topicPostData['IsTruncated']) {
             $postMessage .= '...';
         }
-        $postAuthor = $topicPostData['Author'];
+        
+        $authorId = $topicPostData['author_id'];
```
Comment (Jamiras) [inline resources/views/pages-legacy/forumposthistory.blade.php:70]:
```
Undefined array key "author_id"
```

File: app/Helpers/database/forum.php [hunk 3]
```diff
@@ -3,7 +3,9 @@
 use App\Community\Enums\ArticleType;
 use App\Community\Enums\SubscriptionSubjectType;
 use App\Enums\Permissions;
+use App\Models\ForumTopic;
 use App\Models\Game;
+use App\Models\User;
```
Comment (Jamiras) [inline app/Helpers/database/forum.php:8]:
```
Unknown column 'Author' in 'field list'
```
error on line 182 trying to create a new forum topic

File: app/Helpers/database/user-permission.php (hunk 1 above)
Comment (wescopeland) [reply app/Helpers/database/user-permission.php]: Thanks, fixed in latest.

File: app/Helpers/database/forum.php (hunk 3 above)
Comment (wescopeland) [reply app/Helpers/database/forum.php:8]: Resolved in latest.

File: resources/views/pages-legacy/forumposthistory.blade.php (hunk 2 above)
Comment (wescopeland) [reply resources/views/pages-legacy/forumposthistory.blade.php:71]: Thanks, I wasn't seeing this during testing. It took a little bit of trial and error but I managed to repro. Resolved in latest.

**Review by Jamiras — APPROVED** [review]

**Pull request merged**